game-core/src/test/java/games/strategy/triplea/delegate/battle/steps/BattleStepsTest.java

```diff
@@ -139,6 +145,19 @@ public static Unit givenUnitAttackerFirstStrikeCanNotBeTargetedBy(final UnitType
     return unitAndAttachment.getFirst();
   }
 
+  public static Unit givenUnitCanEvadeAndCanNotBeTargetedBy(final UnitType otherType) {
+    final Tuple<Unit, UnitAttachment> unitAndAttachment = newUnitAndAttachment();
+    when(unitAndAttachment.getSecond().getCanEvade()).thenReturn(true);
+    when(unitAndAttachment.getSecond().getCanNotBeTargetedBy()).thenReturn(Set.of(otherType));
+    return unitAndAttachment.getFirst();
+  }
+
+  public static Unit givenUnitCanNotBeTargetedBy(final UnitType otherType) {
+    final Tuple<Unit, UnitAttachment> unitAndAttachment = newUnitAndAttachment();
+    when(unitAndAttachment.getSecond().getCanNotBeTargetedBy()).thenReturn(Set.of(otherType));
+    return unitAndAttachment.getFirst();
+  }
+
   public static Unit givenUnitDefenderFirstStrike() {
     final UnitType canNotTargetType = mock(UnitType.class);
     final Tuple<Unit, UnitAttachment> unitAndAttachment = newUnitAndAttachment();
```

Comment:
Side note, if you could follow-up and remove the `Tuple` return value from `newUnitAndAttachment`, it would be appreciated. We really do want to delete that class outright in favor of value objects.